interfaces/sourcegen/sourcegen/clib/_CLibSourceGenerator.py

```diff
@@ -267,38 +277,48 @@ def shared_object(cxx_type) -> str:
             cxx_ix += 1
 
         # Obtain class and getter for managed objects
-        uses = [(shared_object(uu.ret_type), uu.name) for uu in c_func.uses]
-        bases |= {uu[0] for uu in uses if uu[0]}
+        shared = []
+        checks = []
+        for uu in c_func.uses:
+            obj = shared_object(uu.ret_type)
+            if obj:
+                shared.append((obj, uu.name))
+                bases |= {obj}
+            else:
+                checks.append(uu.name)
 
         # Ensure that all error codes are set correctly
         error = [-1, "ERR"]
-        cxx_type = cxx_func.ret_type
-        if cxx_type.endswith("int") or cxx_type.endswith("size_t"):
+        cxx_rtype = cxx_func.ret_type
+        cxx_rbase = None
+        if cxx_rtype.endswith("int") or cxx_rtype.endswith("size_t"):
```

Comment:
```suggestion
        if cxx_rtype.endswith(("int", "size_t")):
```
`.startswith()` and `.endswith()` support a tuple for multiple comparisons